src/compiler/parser.ts

```diff
@@ -3254,7 +3268,7 @@ namespace ts {
         function parseParenthesizedArrowFunctionExpressionHead(allowAmbiguity: boolean): ArrowFunction {
             const node = <ArrowFunction>createNode(SyntaxKind.ArrowFunction);
             node.modifiers = parseModifiersForArrowFunction();
-            const isAsync = !!(getModifierFlags(node) & ModifierFlags.Async);
+            const isAsync = (getModifierFlags(node) & ModifierFlags.Async) ? SignatureContext.Await : 0;

```

Comment:
Rather than `0`, could you add a `SignatureContext.None` member of value `0` and use it here? We do this with many of our other enums. Helps with finding references, refactoring and such.